homeassistant/components/homee/cover.py

```diff
@@ -0,0 +1,258 @@
+        """Return the cover's position."""
+        # Translate the homee position values to HA's 0-100 scale
+        if self.has_attribute(AttributeType.POSITION):
+            attribute = self.get_attribute(AttributeType.POSITION)
+            homee_min = attribute.minimum
+            homee_max = attribute.maximum
+            homee_position = attribute.current_value
+            position = ((homee_position - homee_min) / (homee_max - homee_min)) * 100
+
+            return 100 - position
+
+        return None
+
+    @property
+    def current_cover_tilt_position(self) -> int | None:
+        """Return the cover's tilt position."""
+        # Translate the homee position values to HA's 0-100 scale
+        if self.has_attribute(AttributeType.SHUTTER_SLAT_POSITION):
+            attribute = self.get_attribute(AttributeType.SHUTTER_SLAT_POSITION)
+            homee_min = attribute.minimum
+            homee_max = attribute.maximum
+            homee_position = attribute.current_value
+            position = ((homee_position - homee_min) / (homee_max - homee_min)) * 100
+
+            return 100 - position
+
+        return None
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return the opening status of the cover."""
+        if self._open_close_attribute is not None:
+            return (
+                self.attribute(self._open_close_attribute) == 3
+                if not self.is_reversed(self._open_close_attribute)
+                else self.attribute(self._open_close_attribute) == 4
+            )
+
+        return None
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return the closing status of the cover."""
+        if self._open_close_attribute is not None:
+            return (
+                self.attribute(self._open_close_attribute) == 4
+                if not self.is_reversed(self._open_close_attribute)
+                else self.attribute(self._open_close_attribute) == 3
+            )
+
+        return None
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return the state of the cover."""
+        if self.has_attribute(AttributeType.POSITION):
+            return (
+                self.attribute(AttributeType.POSITION)
+                == self.get_attribute(AttributeType.POSITION).maximum
+            )
+
+        if self._open_close_attribute is not None:
+            if not self.is_reversed(self._open_close_attribute):
+                return self.attribute(self._open_close_attribute) == 1
+
+            return self.attribute(self._open_close_attribute) == 0
+
+        # If none of the above is present, it might be a slat only cover.
+        if self.has_attribute(AttributeType.SHUTTER_SLAT_POSITION):
+            return (
+                self.attribute(AttributeType.SHUTTER_SLAT_POSITION)
+                == self.get_attribute(AttributeType.SHUTTER_SLAT_POSITION).minimum
+            )
+
+        return None
```

Comment:
Would it make sense to overwrite `_on_node_updated` to just set the `_attr_is_opening` etc? I think you can deduplicate quite some logic this way

Reply:
I tried and I think it would impair the understandability of the code, since it is quite complicated anyhow due to the reversed-logic.
For `is_closed` I need to keep it this way anyway, since the logic is mor than just reading the `open_close_attribute`.